client/trino-jdbc/src/test/java/io/trino/jdbc/TestJdbcConnection.java

```diff
@@ -501,24 +510,29 @@ public void testConcurrentCancellationOnConnectionClose(boolean autoCommit)
         connection.setAutoCommit(autoCommit);
         futures = range(0, 10)
                 .mapToObj(i -> executor.submit(() -> {
-                    try (Statement statement = connection.createStatement()) {
-                        statement.execute(sql);
+                    try (Statement statement = connection.createStatement();
+                            ResultSet resultSet = statement.executeQuery(sql)) {
+                        //noinspection StatementWithEmptyBody
+                        while (resultSet.next()) {
+                            // consume results
+                        }
                     }
                     return null;
                 }))
                 .collect(toImmutableList());
 
         // Wait for the queries to be started
-        assertEventually(() -> assertThat(listQueryStatuses(sql))
-                .hasSize(futures.size()));
+        assertEventually(() -> {
+            futures.forEach(TestJdbcConnection::assertThatFutureIsBlocked);
+            assertThat(listQueryStatuses(sql))
+                    .hasSize(futures.size());
+        });
 
         // Closing connection should cancel queries
         connection.close();
 
         // verify that all queries were cancelled
-        futures.forEach(future -> assertThatThrownBy(future::get)
-                .hasCauseInstanceOf(SQLException.class)
-                .hasStackTraceContaining("Error executing query"));
+        futures.forEach(future -> assertThatThrownBy(future::get));
```

Comment:
`assertThatThrownBy`... what?
seems like assertion is misisng.

Reply:
`assertThatThrownBy` verifies that exception was throw without checking exception details.